redash/handlers/query_results.py

```diff
@@ -85,7 +87,7 @@ def run_query(query, parameters, data_source, query_id, max_age=0):
     if max_age == 0:
         query_result = None
     else:
-        query_result = models.QueryResult.get_latest(data_source, query.text, max_age)
+        query_result = models.QueryResult.get_latest(data_source, query_text, max_age, apply_auto_limit)
```

Comment:
You already apply auto limit above, so no need to apply it again in `get_latest`. Just like `gen_query_hash`, `get_latest` shouldn't care about applying a limit -- it just get a query text and return a cached version for it, if available.